radar-jersey/src/main/kotlin/org/radarbase/jersey/service/managementportal/MPProjectService.kt

```diff
@@ -56,13 +57,29 @@ class MPProjectService(
         )
 
         organizations = CachedMap(cacheConfig) {
-            mpClient.requestOrganizations()
-                .associateBy { it.id }
-                .also { logger.debug("Fetched organizations {}", it) }
+            try {
+                mpClient.requestOrganizations()
+                    .associateBy { it.id }
+                    .also { logger.debug("Fetched organizations {}", it) }
+            } catch (ex: IOException) {
+                if (ex.message?.contains("404 Not Found") == true) {
+                    logger.warn("Target ManagementPortal does not support organizations. Using default organization main.")
+                    mapOf("main" to defaultOrganization)
+                } else {
+                    throw ex
+                }
+            }
         }
 
         projects = CachedMap(cacheConfig) {
             mpClient.requestProjects()
+                .map { project ->
+                    if (project.organization == null) {
+                        project.copy(organization = defaultOrganization)
```

Comment:
I remember there's also an `organization` key in the old MP versions. Would we also map those to `main` organization?

Reply:
ah good point... at that point it would probably throw a deserialisation error. Let me check.

Reply:
Ok I see, thanks for the updates!